subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -76,6 +76,8 @@
     private final String displayName;
     private final TaskNodeFactory taskNodeFactory;
     private final TaskDependencyResolver dependencyResolver;
+    private final NodeValidator nodeValidator;
+    private final ResourceLock invalidNodeRunningLock;
```

Comment:
Why do you use a `ResourceLock` here and not a simple boolean?

Reply:
That's actually a very good point. I guess claiming "historical reasons" on a draft PR is not realistic, right? :) Anyway, using a `boolean` now.